src/critics/obstacles_critic.cpp

```diff
@@ -61,26 +61,26 @@ void ObstaclesCritic::score(CriticData & data)
     return;
   }
 
+  const size_t traj_len = data.trajectories.x.shape(1);
```

Comment:
Try reverting the obstacle critic changes and see if the smoothness impact is important now still to justify keeping